### PR TITLE
Setup OAuth

### LINKED ISSUE #1: Setup OAuth
- [x] Refacs and cleaning
- [ ] Google ( For Calendar )
- [x] Github ( For Feed and note generation and search )
- [x] Discord (For Feed and search )

### PR DESCRIPTION
**Pull request by kunal-kushwaha**
Passport Template for authentication #1

### CODE REVIEW

File: src/app2.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const express = require('express');
```
Comment (sauravhiremath) [inline src/app2.js]: `app2.js`?

File: src/app2.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+const mongoose = require('mongoose');
```
Comment (sauravhiremath) [inline src/app2.js]: Also, can we separate the server code from the `src` folder?

Something like:

```
.
|-- public/
|-- src/
|-- server/
|-- package.json
|-- ...
```

Need to deploy the server separately

**Review by sauravhiremath — COMMENTED** [review]
Reviewed. Needs some changes

File: server/models/User.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const UserSchema = new Schema({
```
Comment (darshkpatel) [inline server/models/User.js:3]: GitHub username? 
Being used quite a bit

**Review by darshkpatel — CHANGES_REQUESTED** [review]

File: server/models/User.js (hunk 3 above)
Comment (kunal-kushwaha) [reply server/models/User.js:3]: I was thinking name could be that

**Review by darshkpatel — APPROVED** [review]

**Pull request merged**